### PR TITLE
rpm: update macros used in spec

### PR DESCRIPTION
**Pull request by thaJeztah**
Replace some for their equivalent macros. See the [rpm docs for Fedora][1]

- `%{_builddir}` is the equivalent of `%{_topdir}/BUILD`
- `%{_sourcedir}` is the equivalent of `%{_topdir}/SOURCES`

[1]: https://docs.fedoraproject.org/en-US/packaging-guidelines/RPMMacros/#_macros_set_for_the_rpm_and_srpm_build_process

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
Interesting; looks like rawhide builds with this PR? (one failure on `main` for unknown reason); let me try again!

**Comment by thaJeztah** [comment]
Looks like rawhide now passes on x86, but fails on arm64 because `ld` is missing?

```
#21 1.097 + cd /root/rpmbuild/BUILD/containerd.io-0.20240612.144728~0de7043-build
#21 1.097 + make man
#21 2.033 + bin/gen-manpages
#21 2.033 go build  -gcflags=-trimpath=/go/src -buildmode=pie  -o bin/gen-manpages -ldflags '-X github.com/containerd/containerd/v2/version.Version=0.20240612.144728~0de7043 -X github.com/containerd/containerd/v2/version.Revision=0de7043c3a8009c27cfce93320d76f9d408322ce -X github.com/containerd/containerd/v2/version.Package=github.com/containerd/containerd/v2 -s -w ' -tags ""  ./cmd/gen-manpages
#21 17.84 # github.com/containerd/containerd/v2/cmd/gen-manpages
#21 17.84 /usr/local/go/pkg/tool/linux_arm64/link: running gcc failed: exit status 1
#21 17.84 collect2: fatal error: cannot find ‘ld’
#21 17.84 compilation terminated.
#21 17.84 
#21 17.96 make: *** [Makefile:263: bin/gen-manpages] Error 1
#21 17.97 error: Bad exit status from /var/tmp/rpm-tmp.WFgFAK (%build)
#21 17.97     line 40: It's not recommended to have unversioned Obsoletes: Obsoletes: containerd
#21 17.97     line 41: It's not recommended to have unversioned Obsoletes: Obsoletes: runc
#21 17.97     Bad exit status from /var/tmp/rpm-tmp.WFgFAK (%build)
#21 17.97 
#21 17.97 RPM build warnings:
#21 17.97 
#21 17.97 RPM build errors:
#21 ERROR: process "/bin/sh -c /root/build-rpm" did not complete successfully: exit code: 1
```

**Review by dmcgowan — APPROVED** [review]

**Review by vvoland — APPROVED** [review]
